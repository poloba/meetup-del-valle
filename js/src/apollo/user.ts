import { ICurrentUserRole } from "@/types/enums";
import { ApolloCache } from "apollo-cache";
import { NormalizedCacheObject } from "apollo-cache-inmemory";
import { Resolvers } from "apollo-client/core/types";

export default function buildCurrentUserResolver(
  cache: ApolloCache<NormalizedCacheObject>
): Resolvers {
  cache.writeData({
    data: {
      currentUser: {
        __typename: "CurrentUser",
        id: null,
        email: null,
        isLoggedIn: false,
        role: ICurrentUserRole.USER,
      },
      currentActor: {
        __typename: "CurrentActor",
        id: null,
        preferredUsername: null,
        name: null,
        avatar: null,
      },
    },
  });

  return {
    Mutation: {
      updateCurrentUser: (
        _: any,
        {
          id,
          email,
          isLoggedIn,
          role,
        }: { id: string; email: string; isLoggedIn: boolean; role: string },
        { cache: localCache }: { cache: ApolloCache<NormalizedCacheObject> }
      ) => {
        const data = {
          currentUser: {
            id,
            email,
            isLoggedIn,
            role,
            __typename: "CurrentUser",
          },
        };

        localCache.writeData({ data });
      },
      updateCurrentActor: (
        _: any,
        {
          id,
          preferredUsername,
          avatar,
          name,
        }: {
          id: string;
          preferredUsername: string;
          avatar: string;
          name: string;
        },
        { cache: localCache }: { cache: ApolloCache<NormalizedCacheObject> }
      ) => {
        const data = {
          currentActor: {
            id,
            preferredUsername,
            avatar,
            name,
            __typename: "CurrentActor",
          },
        };

        localCache.writeData({ data });
      },
    },
  };
}
