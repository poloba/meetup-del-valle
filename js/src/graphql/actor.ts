import gql from "graphql-tag";

export const FETCH_PERSON = gql`
  query($username: String!) {
    fetchPerson(preferredUsername: $username) {
      id
      url
      name
      domain
      summary
      preferredUsername
      suspended
      mediaSize
      avatar {
        id
        name
        url
      }
      banner {
        id
        url
      }
      feedTokens {
        token
      }
      organizedEvents {
        total
        elements {
          id
          uuid
          title
          beginsOn
        }
      }
    }
  }
`;

export const GET_PERSON = gql`
  query(
    $actorId: ID!
    $organizedEventsPage: Int
    $organizedEventsLimit: Int
    $participationPage: Int
    $participationLimit: Int
  ) {
    person(id: $actorId) {
      id
      url
      name
      domain
      summary
      preferredUsername
      suspended
      mediaSize
      avatar {
        id
        name
        url
      }
      banner {
        id
        url
      }
      feedTokens {
        token
      }
      organizedEvents(
        page: $organizedEventsPage
        limit: $organizedEventsLimit
      ) {
        total
        elements {
          id
          uuid
          title
          beginsOn
        }
      }
      participations(page: $participationPage, limit: $participationLimit) {
        total
        elements {
          id
          event {
            id
            uuid
            title
            beginsOn
          }
        }
      }
      user {
        id
        email
      }
    }
  }
`;

export const LIST_PROFILES = gql`
  query ListProfiles(
    $preferredUsername: String
    $name: String
    $domain: String
    $local: Boolean
    $suspended: Boolean
    $page: Int
    $limit: Int
  ) {
    persons(
      preferredUsername: $preferredUsername
      name: $name
      domain: $domain
      local: $local
      suspended: $suspended
      page: $page
      limit: $limit
    ) {
      total
      elements {
        id
        preferredUsername
        domain
        name
        avatar {
          id
          url
        }
      }
    }
  }
`;

export const UPDATE_DEFAULT_ACTOR = gql`
  mutation ChangeDefaultActor($preferredUsername: String!) {
    changeDefaultActor(preferredUsername: $preferredUsername) {
      id
      defaultActor {
        id
      }
    }
  }
`;

export const CURRENT_ACTOR_CLIENT = gql`
  query currentActor {
    currentActor @client {
      id
      avatar {
        id
        url
      }
      preferredUsername
      name
    }
  }
`;

export const UPDATE_CURRENT_ACTOR_CLIENT = gql`
  mutation UpdateCurrentActor(
    $id: String!
    $avatar: String
    $preferredUsername: String!
    $name: String!
  ) {
    updateCurrentActor(
      id: $id
      avatar: $avatar
      preferredUsername: $preferredUsername
      name: $name
    ) @client
  }
`;

export const LOGGED_USER_PARTICIPATIONS = gql`
  query LoggedUserParticipations(
    $afterDateTime: DateTime
    $beforeDateTime: DateTime
    $page: Int
    $limit: Int
  ) {
    loggedUser {
      id
      participations(
        afterDatetime: $afterDateTime
        beforeDatetime: $beforeDateTime
        page: $page
        limit: $limit
      ) {
        total
        elements {
          event {
            id
            uuid
            title
            picture {
              id
              url
              alt
            }
            beginsOn
            visibility
            organizerActor {
              id
              preferredUsername
              name
              domain
              summary
              avatar {
                id
                url
              }
            }
            attributedTo {
              avatar {
                id
                url
              }
              preferredUsername
              name
              summary
              domain
              url
              id
            }
            participantStats {
              going
              notApproved
              participant
            }
            options {
              maximumAttendeeCapacity
              remainingAttendeeCapacity
            }
            tags {
              id
              slug
              title
            }
          }
          id
          role
          actor {
            id
            preferredUsername
            name
            domain
            summary
            avatar {
              id
              url
            }
          }
        }
      }
    }
  }
`;

export const LOGGED_USER_DRAFTS = gql`
  query LoggedUserDrafts($page: Int, $limit: Int) {
    loggedUser {
      id
      drafts(page: $page, limit: $limit) {
        id
        uuid
        title
        picture {
          id
          url
          alt
        }
        beginsOn
        visibility
        organizerActor {
          id
          preferredUsername
          name
          domain
          avatar {
            id
            url
          }
        }
        participantStats {
          going
          notApproved
        }
        options {
          maximumAttendeeCapacity
          remainingAttendeeCapacity
        }
      }
    }
  }
`;

export const LOGGED_USER_MEMBERSHIPS = gql`
  query LoggedUserMemberships($page: Int, $limit: Int) {
    loggedUser {
      id
      memberships(page: $page, limit: $limit) {
        total
        elements {
          id
          role
          actor {
            id
            avatar {
              id
              url
            }
            preferredUsername
            name
            domain
          }
          parent {
            id
            preferredUsername
            domain
            name
            avatar {
              id
              url
            }
            organizedEvents {
              elements {
                id
                title
                picture {
                  id
                  url
                }
              }
              total
            }
          }
          invitedBy {
            id
            preferredUsername
            domain
            name
            avatar {
              id
              url
            }
          }
        }
      }
    }
  }
`;

export const IDENTITIES = gql`
  query {
    identities {
      id
      avatar {
        id
        url
      }
      preferredUsername
      name
    }
  }
`;

export const PERSON_MEMBERSHIPS = gql`
  query PersonMemberships($id: ID!) {
    person(id: $id) {
      id
      memberships {
        total
        elements {
          id
          role
          parent {
            id
            preferredUsername
            name
            domain
            avatar {
              id
              url
            }
          }
          invitedBy {
            id
            preferredUsername
            name
          }
          insertedAt
          updatedAt
        }
      }
    }
  }
`;

export const PERSON_MEMBERSHIPS_WITH_MEMBERS = gql`
  query PersonMembershipsWithMembers($id: ID!) {
    person(id: $id) {
      id
      memberships {
        total
        elements {
          id
          role
          parent {
            id
            preferredUsername
            name
            domain
            avatar {
              id
              url
            }
            members {
              total
              elements {
                id
                role
                actor {
                  id
                  preferredUsername
                  name
                  domain
                  avatar {
                    id
                    url
                  }
                }
              }
            }
          }
          invitedBy {
            id
            preferredUsername
            name
          }
          insertedAt
          updatedAt
        }
      }
    }
  }
`;

export const CREATE_PERSON = gql`
  mutation CreatePerson(
    $preferredUsername: String!
    $name: String!
    $summary: String
    $avatar: MediaInput
  ) {
    createPerson(
      preferredUsername: $preferredUsername
      name: $name
      summary: $summary
      avatar: $avatar
    ) {
      id
      preferredUsername
      name
      summary
      avatar {
        id
        url
      }
    }
  }
`;

export const UPDATE_PERSON = gql`
  mutation UpdatePerson(
    $id: ID!
    $name: String
    $summary: String
    $avatar: MediaInput
  ) {
    updatePerson(id: $id, name: $name, summary: $summary, avatar: $avatar) {
      id
      preferredUsername
      name
      summary
      avatar {
        id
        url
      }
    }
  }
`;

export const DELETE_PERSON = gql`
  mutation DeletePerson($id: ID!) {
    deletePerson(id: $id) {
      preferredUsername
    }
  }
`;

/**
 * This one is used only to register the first account.
 * Prefer CREATE_PERSON when creating another identity
 */
export const REGISTER_PERSON = gql`
  mutation(
    $preferredUsername: String!
    $name: String!
    $summary: String!
    $email: String!
  ) {
    registerPerson(
      preferredUsername: $preferredUsername
      name: $name
      summary: $summary
      email: $email
    ) {
      id
      preferredUsername
      name
      summary
      avatar {
        id
        url
      }
    }
  }
`;

export const SUSPEND_PROFILE = gql`
  mutation SuspendProfile($id: ID!) {
    suspendProfile(id: $id) {
      id
    }
  }
`;

export const UNSUSPEND_PROFILE = gql`
  mutation UnSuspendProfile($id: ID!) {
    unsuspendProfile(id: $id) {
      id
    }
  }
`;
