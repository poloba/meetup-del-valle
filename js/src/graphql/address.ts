import gql from "graphql-tag";

const $addressFragment = `
id,
description,
geom,
street,
locality,
postalCode,
region,
country,
type,
url,
originId
`;

export const ADDRESS = gql`
    query($query:String!, $locale: String) {
        searchAddress(
            query: $query,
            locale: $locale
        ) {
            ${$addressFragment}
        }
    }
`;

export const REVERSE_GEOCODE = gql`
    query($latitude: Float!, $longitude: Float!, $zoom: Int, $locale: String) {
        reverseGeocode(latitude: $latitude, longitude: $longitude, zoom: $zoom, locale: $locale) {
            ${$addressFragment}
        }
    }
`;
