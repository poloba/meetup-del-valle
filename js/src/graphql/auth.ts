import gql from "graphql-tag";

export const LOGIN = gql`
  mutation Login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      accessToken
      refreshToken
      user {
        id
        email
        role
      }
    }
  }
`;

export const SEND_RESET_PASSWORD = gql`
  mutation SendResetPassword($email: String!) {
    sendResetPassword(email: $email)
  }
`;

export const RESET_PASSWORD = gql`
  mutation ResetPassword($token: String!, $password: String!) {
    resetPassword(token: $token, password: $password) {
      accessToken
      refreshToken
      user {
        id
      }
    }
  }
`;

export const RESEND_CONFIRMATION_EMAIL = gql`
  mutation ResendConfirmationEmail($email: String!) {
    resendConfirmationEmail(email: $email)
  }
`;

export const REFRESH_TOKEN = gql`
  mutation RefreshToken($refreshToken: String!) {
    refreshToken(refreshToken: $refreshToken) {
      accessToken
      refreshToken
    }
  }
`;
