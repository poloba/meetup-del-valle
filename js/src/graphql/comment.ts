import gql from "graphql-tag";

export const COMMENT_FIELDS_FRAGMENT_NAME = "CommentFields";
export const COMMENT_FIELDS_FRAGMENT = gql`
  fragment CommentFields on Comment {
    id
    uuid
    url
    text
    visibility
    local
    actor {
      avatar {
        id
        url
      }
      id
      domain
      preferredUsername
      name
      summary
    }
    totalReplies
    updatedAt
    deletedAt
  }
`;

export const COMMENT_RECURSIVE_FRAGMENT = gql`
  fragment CommentRecursive on Comment {
    ...CommentFields
    inReplyToComment {
      ...CommentFields
    }
    originComment {
      ...CommentFields
    }
    replies {
      ...CommentFields
      replies {
        ...CommentFields
      }
    }
  }
  ${COMMENT_FIELDS_FRAGMENT}
`;

export const FETCH_THREAD_REPLIES = gql`
  query ($threadId: ID!) {
    thread(id: $threadId) {
      ...CommentRecursive
    }
  }
  ${COMMENT_RECURSIVE_FRAGMENT}
`;

export const COMMENTS_THREADS = gql`
  query ($eventUUID: UUID!) {
    event(uuid: $eventUUID) {
      id
      uuid
      comments {
        ...CommentFields
      }
    }
  }
  ${COMMENT_FIELDS_FRAGMENT}
`;

export const CREATE_COMMENT_FROM_EVENT = gql`
  mutation CreateCommentFromEvent(
    $eventId: ID!
    $text: String!
    $inReplyToCommentId: ID
  ) {
    createComment(
      eventId: $eventId
      text: $text
      inReplyToCommentId: $inReplyToCommentId
    ) {
      ...CommentRecursive
    }
  }
  ${COMMENT_RECURSIVE_FRAGMENT}
`;

export const DELETE_COMMENT = gql`
  mutation DeleteComment($commentId: ID!) {
    deleteComment(commentId: $commentId) {
      id
    }
  }
`;

export const UPDATE_COMMENT = gql`
  mutation UpdateComment($commentId: ID!, $text: String!) {
    updateComment(commentId: $commentId, text: $text) {
      ...CommentFields
    }
  }
  ${COMMENT_FIELDS_FRAGMENT}
`;
