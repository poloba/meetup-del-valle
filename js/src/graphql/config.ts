import gql from "graphql-tag";

export const CONFIG = gql`
  query {
    config {
      name
      description
      slogan
      registrationsOpen
      registrationsAllowlist
      demoMode
      countryCode
      languages
      anonymous {
        participation {
          allowed
          validation {
            email {
              enabled
              confirmationRequired
            }
            captcha {
              enabled
            }
          }
        }
        eventCreation {
          allowed
          validation {
            email {
              enabled
              confirmationRequired
            }
            captcha {
              enabled
            }
          }
        }
        reports {
          allowed
        }
        actorId
      }
      location {
        latitude
        longitude
        # accuracyRadius
      }
      maps {
        tiles {
          endpoint
          attribution
        }
        routing {
          type
        }
      }
      geocoding {
        provider
        autocomplete
      }
      resourceProviders {
        type
        endpoint
        software
      }
      features {
        groups
        eventCreation
        koenaConnect
      }
      auth {
        ldap
        oauthProviders {
          id
          label
        }
      }
      uploadLimits {
        default
        avatar
        banner
      }
      instanceFeeds {
        enabled
      }
    }
  }
`;

export const TERMS = gql`
  query Terms($locale: String) {
    config {
      terms(locale: $locale) {
        type
        url
        bodyHtml
      }
    }
  }
`;

export const ABOUT = gql`
  query About {
    config {
      name
      description
      longDescription
      contact
      languages
      registrationsOpen
      registrationsAllowlist
      anonymous {
        participation {
          allowed
        }
      }
      version
      federating
      instanceFeeds {
        enabled
      }
    }
  }
`;

export const CONTACT = gql`
  query Contact {
    config {
      name
      contact
    }
  }
`;

export const RULES = gql`
  query Rules {
    config {
      rules
    }
  }
`;

export const PRIVACY = gql`
  query Privacy($locale: String) {
    config {
      privacy(locale: $locale) {
        type
        url
        bodyHtml
      }
    }
  }
`;

export const TIMEZONES = gql`
  query {
    config {
      timezones
    }
  }
`;
