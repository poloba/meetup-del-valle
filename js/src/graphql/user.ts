import gql from "graphql-tag";

export const CREATE_USER = gql`
  mutation CreateUser($email: String!, $password: String!, $locale: String) {
    createUser(email: $email, password: $password, locale: $locale) {
      email
      confirmationSentAt
    }
  }
`;

export const VALIDATE_USER = gql`
  mutation ValidateUser($token: String!) {
    validateUser(token: $token) {
      accessToken
      refreshToken
      user {
        id
        email
        defaultActor {
          id
          preferredUsername
          name
          avatar {
            id
            url
          }
        }
      }
    }
  }
`;

export const LOGGED_USER = gql`
  query {
    loggedUser {
      id
      email
      defaultActor {
        id
        preferredUsername
        name
        avatar {
          id
          url
        }
      }
      provider
    }
  }
`;

export const CHANGE_PASSWORD = gql`
  mutation ChangePassword($oldPassword: String!, $newPassword: String!) {
    changePassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      id
    }
  }
`;

export const CHANGE_EMAIL = gql`
  mutation ChangeEmail($email: String!, $password: String!) {
    changeEmail(email: $email, password: $password) {
      id
    }
  }
`;

export const VALIDATE_EMAIL = gql`
  mutation ValidateEmail($token: String!) {
    validateEmail(token: $token) {
      id
    }
  }
`;

export const DELETE_ACCOUNT = gql`
  mutation DeleteAccount($password: String, $userId: ID) {
    deleteAccount(password: $password, userId: $userId) {
      id
    }
  }
`;

export const SUSPEND_USER = gql`
  mutation SuspendUser($userId: ID) {
    deleteAccount(userId: $userId) {
      id
    }
  }
`;

export const CURRENT_USER_CLIENT = gql`
  query {
    currentUser @client {
      id
      email
      isLoggedIn
      role
    }
  }
`;

export const UPDATE_CURRENT_USER_CLIENT = gql`
  mutation UpdateCurrentUser(
    $id: String!
    $email: String!
    $isLoggedIn: Boolean!
    $role: UserRole!
  ) {
    updateCurrentUser(
      id: $id
      email: $email
      isLoggedIn: $isLoggedIn
      role: $role
    ) @client
  }
`;

export const USER_SETTINGS_FRAGMENT = gql`
  fragment UserSettingFragment on UserSettings {
    timezone
    notificationOnDay
    notificationEachWeek
    notificationBeforeEvent
    notificationPendingParticipation
    notificationPendingMembership
    location {
      range
      geohash
      name
    }
  }
`;

export const USER_SETTINGS = gql`
  query UserSetting {
    loggedUser {
      id
      locale
      settings {
        ...UserSettingFragment
      }
    }
  }
  ${USER_SETTINGS_FRAGMENT}
`;

export const SET_USER_SETTINGS = gql`
  mutation SetUserSettings(
    $timezone: String
    $notificationOnDay: Boolean
    $notificationEachWeek: Boolean
    $notificationBeforeEvent: Boolean
    $notificationPendingParticipation: NotificationPendingEnum
    $notificationPendingMembership: NotificationPendingEnum
    $location: LocationInput
  ) {
    setUserSettings(
      timezone: $timezone
      notificationOnDay: $notificationOnDay
      notificationEachWeek: $notificationEachWeek
      notificationBeforeEvent: $notificationBeforeEvent
      notificationPendingParticipation: $notificationPendingParticipation
      notificationPendingMembership: $notificationPendingMembership
      location: $location
    ) {
      ...UserSettingFragment
    }
  }
  ${USER_SETTINGS_FRAGMENT}
`;

export const LIST_USERS = gql`
  query ListUsers($email: String, $page: Int, $limit: Int) {
    users(email: $email, page: $page, limit: $limit) {
      total
      elements {
        id
        email
        locale
        confirmedAt
        disabled
        actors {
          id
          preferredUsername
          avatar {
            id
            url
          }
          name
          summary
        }
        settings {
          timezone
        }
      }
    }
  }
`;

export const GET_USER = gql`
  query GetUser($id: ID!) {
    user(id: $id) {
      id
      email
      confirmedAt
      confirmationSentAt
      lastSignInAt
      lastSignInIp
      currentSignInIp
      currentSignInAt
      locale
      disabled
      mediaSize
      defaultActor {
        id
      }
      actors {
        id
        preferredUsername
        name
        avatar {
          id
          url
        }
      }
      participations {
        total
      }
      role
    }
  }
`;

export const UPDATE_USER_LOCALE = gql`
  mutation UpdateUserLocale($locale: String!) {
    updateLocale(locale: $locale) {
      id
      locale
    }
  }
`;
