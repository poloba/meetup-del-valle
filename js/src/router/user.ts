import { beforeRegisterGuard } from "@/router/guards/register-guard";
import { Route, RouteConfig } from "vue-router";
import { EsModuleComponent } from "vue/types/options";

export enum UserRouteName {
  REGISTER = "Register",
  REGISTER_PROFILE = "RegisterProfile",
  RESEND_CONFIRMATION = "ResendConfirmation",
  SEND_PASSWORD_RESET = "SendPasswordReset",
  PASSWORD_RESET = "PasswordReset",
  EMAIL_VALIDATE = "EMAIL_VALIDATE",
  VALIDATE = "Validate",
  LOGIN = "Login",
}

export const userRoutes: RouteConfig[] = [
  {
    path: "/register/user",
    name: UserRouteName.REGISTER,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "RegisterUser" */ "@/views/User/Register.vue"
      ),
    props: true,
    meta: { requiredAuth: false },
    beforeEnter: beforeRegisterGuard,
  },
  {
    path: "/register/profile",
    name: UserRouteName.REGISTER_PROFILE,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "RegisterProfile" */ "@/views/Account/Register.vue"
      ),
    // We can only pass string values through params, therefore
    props: (route: Route): Record<string, unknown> => ({
      email: route.params.email,
      userAlreadyActivated: route.params.userAlreadyActivated === "true",
    }),
    meta: { requiredAuth: false },
  },
  {
    path: "/resend-instructions",
    name: UserRouteName.RESEND_CONFIRMATION,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "ResendConfirmation" */ "@/views/User/ResendConfirmation.vue"
      ),
    props: true,
    meta: { requiresAuth: false },
  },
  {
    path: "/password-reset/send",
    name: UserRouteName.SEND_PASSWORD_RESET,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "SendPasswordReset" */ "@/views/User/SendPasswordReset.vue"
      ),
    props: true,
    meta: { requiresAuth: false },
  },
  {
    path: "/password-reset/:token",
    name: UserRouteName.PASSWORD_RESET,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "PasswordReset" */ "@/views/User/PasswordReset.vue"
      ),
    meta: { requiresAuth: false },
    props: true,
  },
  {
    path: "/validate/email/:token",
    name: UserRouteName.EMAIL_VALIDATE,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "EmailValidate" */ "@/views/User/EmailValidate.vue"
      ),
    props: true,
    meta: { requiresAuth: false },
  },
  {
    path: "/validate/:token",
    name: UserRouteName.VALIDATE,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "Validate" */ "@/views/User/Validate.vue"),
    props: true,
    meta: { requiresAuth: false },
  },
  {
    path: "/login",
    name: UserRouteName.LOGIN,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "Login" */ "@/views/User/Login.vue"),
    props: true,
    meta: { requiredAuth: false },
  },
];
