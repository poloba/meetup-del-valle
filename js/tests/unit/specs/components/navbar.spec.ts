import { shallowMount, createLocalVue, Wrapper, config } from "@vue/test-utils";
import NavBar from "@/components/NavBar.vue";
import {
  createMockClient,
  MockApolloClient,
  RequestHandler,
} from "mock-apollo-client";
import VueApollo from "vue-apollo";
import { CONFIG } from "@/graphql/config";
import { USER_SETTINGS } from "@/graphql/user";
import { InMemoryCache } from "apollo-cache-inmemory";
import buildCurrentUserResolver from "@/apollo/user";
import Buefy from "buefy";
import { configMock } from "../mocks/config";

const localVue = createLocalVue();
localVue.use(VueApollo);
localVue.use(Buefy);
config.mocks.$t = (key: string): string => key;

describe("App component", () => {
  let wrapper: Wrapper<Vue>;
  let mockClient: MockApolloClient | null;
  let apolloProvider;
  let requestHandlers: Record<string, RequestHandler>;

  const createComponent = (handlers = {}, baseData = {}) => {
    const cache = new InMemoryCache({ addTypename: true });

    mockClient = createMockClient({
      cache,
      resolvers: buildCurrentUserResolver(cache),
    });

    requestHandlers = {
      configQueryHandler: jest.fn().mockResolvedValue(configMock),
      loggedUserQueryHandler: jest.fn().mockResolvedValue(null),
      ...handlers,
    };

    mockClient.setRequestHandler(CONFIG, requestHandlers.configQueryHandler);

    mockClient.setRequestHandler(
      USER_SETTINGS,
      requestHandlers.loggedUserQueryHandler
    );

    apolloProvider = new VueApollo({
      defaultClient: mockClient,
    });

    wrapper = shallowMount(NavBar, {
      localVue,
      apolloProvider,
      stubs: ["router-link", "router-view"],
      data() {
        return {
          ...baseData,
        };
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    mockClient = null;
    apolloProvider = null;
  });

  it("renders a Vue component", async () => {
    createComponent();

    await wrapper.vm.$nextTick();

    expect(wrapper.exists()).toBe(true);
    expect(requestHandlers.configQueryHandler).toHaveBeenCalled();
    expect(wrapper.vm.$apollo.queries.config).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findComponent({ name: "b-navbar" }).exists()).toBeTruthy();
  });
});
