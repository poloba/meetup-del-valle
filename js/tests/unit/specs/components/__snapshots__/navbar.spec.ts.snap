// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders a Vue component 1`] = `<b-navbar-stub type="is-secondary" wrapperclass="container" closeonclick="true" mobileburger="true"><template></template> <template></template> <template></template></b-navbar-stub>`;
